gtda/homology/point_clouds.py

```diff
@@ -441,19 +429,184 @@ def transform(self, X, y=None):
+        value is declared to be equal to `max_edge_length`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    infinity_values_ : float
+        Effective death value to assign to features which are still alive at
+        filtration value `max_edge_length`.
+
+    See also
+    --------
+    VietorisRipsPersistence, SparseRipsPersistence, ConsistentRescaling
+
+    Notes
+    -----
+    `GUDHI <https://github.com/GUDHI/gudhi-devel>`_ is used as a C++ backend
+    for computing Cech persistent homology. Python bindings were modified
+    for performance.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] C. Maria, "Persistent Cohomology", 2020; `GUDHI User and Reference \
+    Manual <http://gudhi.gforge.inria.fr/doc/3.1.0/group__persistent_\
+    cohomology.html>`_.
+
+    """
+    _hyperparameters = {'max_edge_length': [numbers.Number],
+                        'infinity_values_': [numbers.Number],
+                        '_homology_dimensions': [list, [int, (0, np.inf)]],
+                        'coeff': [int, (2, np.inf)]}
+
+    def __init__(self, max_edge_length=np.inf, homology_dimensions=(0, 1),
+                 coeff=2, infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.max_edge_length = max_edge_length
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _gudhi_diagram(self, X):
+        cech_complex = CechComplex(points=X, max_radius=self.max_edge_length)
+        simplex_tree = cech_complex.create_simplex_tree(
+            max_dimension=max(self._homology_dimensions) + 1)
+        Xdgms = simplex_tree.persistence(
+            homology_coeff_field=self.coeff, min_persistence=0)
+
+        # Separate diagrams by homology dimensions
+        Xdgms = {dim: np.array([Xdgms[i][1] for i in range(len(Xdgms))
+                                if Xdgms[i][0] == dim]).reshape((-1, 2))
+                 for dim in self.homology_dimensions}
+
+        if 0 in self._homology_dimensions:
+            Xdgms[0] = Xdgms[0][1:, :]  # Remove final death at np.inf
+
+        # Add dimension as the third elements of each (b, d) tuple
+        Xdgms = {dim: np.hstack([Xdgms[dim],
+                                 dim * np.ones((Xdgms[dim].shape[0], 1),
+                                               dtype=Xdgms[dim].dtype)])
+                 for dim in self._homology_dimensions}
+        return Xdgms
+
+    def fit(self, X, y=None):
+        """Calculate `infinity_values_`. Then, return the estimator.
```

Comment:
Replace with :attr:`infinity_values_`

Reply:
I took the liberty to use :param: throughout the file as well to identify parameters.

Reply:
We don't use :param: anywhere else, and I recall it not behaving as I wanted. How does it display?

Reply:
Badly.